test/DynamoCoreTests/DelayedGraphExecutionTests.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using Dynamo.Selection;
+using System.Linq;
+using Dynamo.Models;
+using NUnit.Framework;
+using Dynamo.Graph.Workspaces;
+using static Dynamo.Models.DynamoModel;
+using Dynamo.Scheduler;
+
+namespace Dynamo.Tests
+{
+    [TestFixture]
+    class DelayedGraphExecutionTests : DynamoModelTestBase
+    {
+        private HomeWorkspaceModel wModel;
+        private int taskCounter = 0;
+        private bool activeTaskCounter = false;
+        private int originalNodeCount = 0;
+        private int expectedTaskModifiedNodeCount = 0;
+
+        private void activateTaskCounter(int expected)
+        {
+            activeTaskCounter = true;
+            taskCounter = 0;
+            expectedTaskModifiedNodeCount = expected;
+        }
+
+        private void deactivateTaskCounter()
+        {
+            activeTaskCounter = false;
+            taskCounter = 0;
+        }
+
+        private List<Graph.ModelBase> workspaceNodes()
+        {
+            return wModel.Nodes.Select(x => x as Graph.ModelBase).ToList();
+        }
+
+        [SetUp]
+        public void Init()
+        {
+            OpenModel(Path.Combine(TestDirectory, "core", "DelayedGraphExecution.dyn"));
+            wModel = CurrentDynamoModel.CurrentWorkspace as HomeWorkspaceModel;
+            wModel.RunSettings.RunType = RunType.Automatic;
+
+            originalNodeCount = workspaceNodes().Count();
+
+            CurrentDynamoModel.Scheduler.TaskStateChanged += (DynamoScheduler sender, TaskStateChangedEventArgs args) =>
```

Comment:
can you unsubscribe this handler when the test is in teardown?

Reply:
yes